pilot/pkg/networking/core/listener_waypoint.go

```diff
@@ -297,13 +328,20 @@ func (lb *ListenerBuilder) buildWaypointInternal(wls []model.WorkloadInfo, svcs
 			},
 		}
 	}
+
 	for _, svc := range svcs {
+		var waypoint host.Name
+		if isEastWestGateway && features.EnableAmbientMultiNetwork {
```

Comment:
Do we also need to check `EnableAmbientWaypointMultiNetwork` here?

Reply:
I tried to change the code to check the feature flag here, but it ended up a pretty big change.

Basically, we have to preserve the old Envoy configuration when the flag is not set and use the new one when it's set. Two of those codepaths together add up to a reasoanbly large amount of code.

Given that:

1. The new configration is backward compatible with the old one
2. Both code paths apply to ambient E/W gateway only

Unless there is a strong preference to do otherwise, I'd rather keep this code as-is.

Reply:
Fair enough